nengo_loihi/builder/tests/test_connection.py

```diff
@@ -0,0 +1,67 @@
+from distutils.version import LooseVersion
+
+import nengo
+from nengo.exceptions import BuildError
+import numpy as np
+import pytest
+
+
+@pytest.mark.skipif(LooseVersion(nengo.__version__) <= LooseVersion('2.8.0'),
+                    reason="requires more recent Nengo version")
+def test_split_conv2d_transform_error(Simulator):
+    with nengo.Network() as net:
+        node_offchip = nengo.Node([1])
+        ens_onchip = nengo.Ensemble(10, 1)
+        conv2d = nengo.Convolution(
+            n_filters=1, input_shape=(1, 1, 1), kernel_size=(1, 1))
+        nengo.Connection(node_offchip, ens_onchip, transform=conv2d)
+
+    with pytest.raises(BuildError, match="Conv2D"):
+        with Simulator(net):
+            pass
+
+
+@pytest.mark.parametrize("pre_dims", [1, 3])
+@pytest.mark.parametrize("post_dims", [1, 3])
+@pytest.mark.parametrize("learn", [True, False])
+@pytest.mark.parametrize("use_solver", [True, False])
```

Comment:
This is 16 tests. Do we need every possible combination here? They don't take long on the emulator, but I'm worried they could add more time than necessary on the chip.

Reply:
This unit test was copied from the old splitter tests. Perhaps this should be added to a card to investigate/prune the unit tests for CI?

Reply:
Dan already did some of this not too long ago. If this is from before, it's probably already fairly efficient. We can leave it for now, and perhaps in a few months we'll be revisiting tests again and can do some pruning.